### ISSUE TITLE
Make global ARIA attributes configurable in aria-allowed-attr

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2108]
To make axe-core more flexible and allow it to support different environments it would be helpful to add more configuration options. One such option is to be able to set which global attributes are available in aria-allowed-attr, see https://github.com/dequelabs/axe-core/issues/2078 for details.

**Comment by straker** [comment]
qa: to test this you can take a rule such as `aria-allowed-attrs` and test any role in the new [standards object](https://github.com/dequelabs/axe-core/blob/develop/lib/standards/aria-roles.js) that has an `allowedAttrs` property. First test that you can't add an aria-attribute that is not one 1) global 2) part of the `reuqiredAttrs` and 3) part of the `allowedAttrs`, then configure the role to allow the aria-attribute using `axe.configure`.

For example, the [checkbox role](https://github.com/dequelabs/axe-core/blob/develop/lib/standards/aria-roles.js#L59-L68) allows attributes `aria-checked`, `aria-readonly`, and `aria-required`. To test, you can add `aria-expanded` to an element that has `role=checkbox` and see that it should fail `aria-allowed-attrs` rule. Then configure the standards object to allow it and test it again, and it should pass.

```html
<div role="checkbox" aria-expanded="true" aria-checked="true"></div>

<script>
axe.configure({
  standards: {
    // the standards object to configure https://github.com/dequelabs/axe-core/blob/develop/lib/standards
    ariaRoles: {
      // the role to configure
      checkbox: {
        // the property to configure. with arrays you'll need to copy what's in the standards
       //  object and add the new values, otherwise the previous values will be removed
       // (meaning `aria-checked`, etc. would no longer be allowed)
        allowedAttrs: ['aria-checked', 'aria-readonly', 'aria-required', 'aria-expanded']
      }
    }
  }
});
</script>

await axe.run();  // so not show violation for `aria-allowed-attrs` for `aria-expanded` on the checkbox role
```

**Comment by padmavemulapati** [comment]
Verified as per QA notes using latest axe-core(3.5.5 pulled code)
Step1: for the test file ' <div role="checkbox" aria-expanded="true" aria-checked="true">Hello</div>
Simply running axe - would result - failure for `aria-allowed-attr` (attached the Screenshot)
![image](https://user-images.githubusercontent.com/47409579/88069007-89728600-cb8e-11ea-9fb9-576c042f2c83.png)

Step2: configured with `axe.configure({
  standards: {
    ariaRoles: {
     checkbox: {
       allowedAttrs: ['aria-checked', 'aria-readonly', 'aria-required', 'aria-expanded']
      }
    }
  }
});`
and did - axe.run(); - resulting - pass for `aria-allowed-attr` (attached the screenshot)
![image](https://user-images.githubusercontent.com/47409579/88069554-3cdb7a80-cb8f-11ea-9955-89953c215c2d.png)

**Comment by jeankaplansky** [comment]
Included in internal release notes 7/27/2020
